### DISCUSSION TITLE
Can't add Cloudflare SSL into Nginx Proxy Manager

### DISCUSSION THREAD

**Discussion by Klariff** [tteck/Proxmox#1291] [Help Needed]
I have installed the most recent version of the Nginx Proxy Manager script, then I try to add an SSL certificate using a DNS challenge from Cloudflare but this error appears:

certbot: error: unrecognized arguments: --dns-cloudflare-credentials /etc/letsencrypt/credentials/credentials-1

**Comment by tteck** [comment]
What are you calling the "most recent" version? The main branch is locked to version 2.9.22, while the dev branch has the latest version 2.10.2 (which installs `certbot-dns-cloudflare`)

You can also try installing `certbot-dns-cloudflare` 
```
python3 -m pip install --no-cache-dir certbot-dns-cloudflare
```

**Comment by Klariff** [comment]
Sorry I didn't specify, most recent version of the setup script, i'll test installing certbot-dns-cloudflare

**Comment by Klariff** [comment]
Hmm, now appears the following error:

Error: Command failed: certbot certonly --config "/etc/letsencrypt.ini" --cert-name "npm-2" --agree-tos --email "<EMAIL>" --domains "<DOMAIN>" --authenticator dns-cloudflare --dns-cloudflare-credentials "/etc/letsencrypt/credentials/credentials-2"
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Some challenges have failed.
Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /var/log/letsencrypt/letsencrypt.log or re-run Certbot with -v for more details.

    at ChildProcess.exithandler (node:child_process:402:12)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1100:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:304:5)

**Comment by tteck** [comment]
Since I don't use Cloudflare, I'm afraid I can't offer any help regarding it.

**Comment by Koliera** [comment]
I used this tutorial (though shorter expire date) and it works great with the NPM from the script - https://www.youtube.com/watch?v=pwK1LnbTitI
